### ISSUE TITLE
How to update npm in Dev Container?

### ISSUE THREAD

**Issue by richb-hanover** [PRQL/prql#3377]
### What's up?

When starting the Dev Container, I see the following warning about a new minor version (9.8.1) of `npm` being available. I wasn't able to figure out where npm gets installed into the Dev Container...

My request would be to install a later version of npm in a timely way so it doesn't bite us at an unfortunate time. Thanks.

```
[13274 ms] Start: Run in container: /bin/sh -c task install-maturin && task install-pre-commit && pre-commit install-hooks
[13274 ms] Start: Run in container: /bin/sh -c task install-npm-dependencies
Running install-npm-dependencies from devcontainer.json...
task: [install-npm-dependencies] npm install --location=global prettier prettier-plugin-go-template

added 3 packages in 3s

1 package is looking for funding
  run `npm fund` for details
npm notice 
npm notice New minor version of npm available! 9.6.7 -> 9.8.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v9.8.1
npm notice Run npm install -g npm@9.8.1 to update!
npm notice 
```

**Comment by max-sixty** [comment]
Possibly that's here? https://github.com/PRQL/prql/blob/a9bab37522e66172c0353542304bb2ea46c20e4b/.devcontainer/devcontainer.json#L7

But not sure if that's update-able...

If there's a way to systematize this so it's always current, it might be worth doing. Otherwise we can close until there's a symptom — there will always be slightly out-of-date libs...

**Comment by richb-hanover** [comment]
Yes - your link to _.devcontainer/devcontainer.json_ points to the `node` devcontainer. That json file requests version `:1` which gets the latest version.

That devcontainer seems to have been built from [this json file](https://github.com/devcontainers/features/blob/main/src/node/devcontainer-feature.json) that explicitly includes `nvm` version 0.39.2. This matches the installed version in the Dev Container.

Current `nvm` is 0.39.5. A little sleuthing in the [nvm repo](https://github.com/nvm-sh/nvm) makes it look as if 0.39.2 was released in Sept-Oct 2022, so it's not very old.

**TL;DR** @max-sixty is correct - this should update by itself when the upstream devcontainer changes, and we can close this.

**Comment by eitsupi** [comment]
Yes, this is recommended to open the issue to https://github.com/devcontainers/features because node.js is installed through the node Feature (`ghcr.io/devcontainers/features/node`).

**Comment by eitsupi** [comment]
devcontainers/features#666
